src/main/java/com/microsoft/sqlserver/jdbc/SQLServerBulkCSVFileRecord.java

```diff
@@ -229,21 +236,24 @@ public void close() throws SQLServerException {
         if (null == currentLine)
             return null;
         else {
-            // Binary data may be corrupted
-            // The limit in split() function should be a negative value,
-            // otherwise trailing empty strings are discarded.
-            // Empty string is returned if there is no value.
-            String[] data = currentLine.split(delimiter, -1);
-
-            // Cannot go directly from String[] to Object[] and expect it to act
-            // as an array.
+            /*
+             * Binary data may be corrupted The limit in split() function should be a negative value, otherwise trailing
+             * empty strings are discarded. Empty string is returned if there is no value.
+             */
+            String[] data = (escapeDelimiters && currentLine.contains("\""))
+                                                                             ? escapeQuotesRFC4180(currentLine.split(
+                                                                                     delimiter + escapeSplitPattern))
+                                                                             : currentLine.split(delimiter, -1);
+            /*
+             * Cannot go directly from String[] to Object[] and expect it to act as an array.
```

Comment:
[not a big deal, and only if you're changing this file anyway]
this and line 254-256 is only a single line comment